### PR TITLE
device: support managed network with physical nic

### PR DESCRIPTION
**Pull request by gabrielmougard**
Fixes: https://github.com/lxc/lxd/issues/11177

### CODE REVIEW

**Comment by tomponline** [comment]
This is failing static analysis

**Comment by tomponline** [comment]
[/golangci-lint/issues/2649.](https://github.com/golangci/golangci-lint/issues/2649.)"
lxd/device/nic_physical.go:293: File is not `gofmt`-ed with `-s` (gofmt)

**Comment by tomponline** [comment]
We require all files to be run through go fmt

File: doc/reference/devices_nic.md [hunk 1]
```diff
@@ -250,17 +250,18 @@ The targeted device will vanish from the host and appear in the instance (which
...
+Key                     | Type    | Default           | Managed | Description
+:--                     | :--     | :--               | :--     | :--
+`boot.priority`         | integer | -                 | no      | Boot priority for VMs (higher value boots first)
+`gvrp`                  | bool    | `false`           | no      | Register VLAN using GARP VLAN Registration Protocol
+`hwaddr`                | string  | randomly assigned | no      | The MAC address of the new interface
+`maas.subnet.ipv4`      | string  | -                 | no      | MAAS IPv4 subnet to register the instance in
+`maas.subnet.ipv6`      | string  | -                 | no      | MAAS IPv6 subnet to register the instance in
+`mtu`                   | integer | parent MTU        | no      | The MTU of the new interface
+`name`                  | string  | kernel assigned   | no      | The name of the interface inside the instance
+`network`               | string  | -                 | no      | The managed network to ink the device to (instead of specifying the `nictype` directly)
```
[the remark this answers is not in the document]
Comment (gabrielmougard) [reply doc/reference/devices_nic.md]: I checked, it's okay

**Review by tomponline — APPROVED** [review]

File: doc/reference/devices_nic.md [hunk 2]
```diff
@@ -250,17 +250,18 @@ The targeted device will vanish from the host and appear in the instance (which
```
Comment (ru-fu) [inline doc/reference/devices_nic.md:250]: Sorry, a bit late with my review - but there's two other small updates needed in this file.

- The note for `nictype`: `physical` (line 240) still states that you can only use `nictype`, not `network`.
- In the "Available NIC types" section, `physical` must now be moved to the section that supports `nictype` and `network`.

Let me know if you'd prefer me to do the changes, but maybe it's a good exercise. :)

Comment (gabrielmougard) [reply doc/reference/devices_nic.md:250]: I fixing that ASAP

**Pull request merged**